src/main/java/org/jabref/logic/importer/util/GrobidService.java

```diff
@@ -0,0 +1,58 @@
+package org.jabref.logic.importer.util;
+
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+
+import org.jabref.logic.net.URLDownload;
+import org.jabref.preferences.JabRefPreferences;
+
+/**
+ * Implements an API to a GROBID server, as described at
+ * https://grobid.readthedocs.io/en/latest/Grobid-service/#grobid-web-services
+ *
+ * Note: Currently a custom GROBID server is used...
+ * https://github.com/NikodemKch/grobid
+ *
+ * The methods are structured to match the GROBID server api.
+ * Each method corresponds to a GROBID service request. Only the ones already used are already implemented.
+ */
+public class GrobidService {
+
+  public enum ConsolidateCitations {
+      NO(0), WITH_METADATA(1), WITH_DOI_ONLY(2);
+      private int code;
+
+      ConsolidateCitations(int code) {
+          this.code = code;
+      }
+
+        public int getCode() {
+            return this.code;
+        }
+    }
+
+    private static JabRefPreferences jabRefPreferences;
```

Comment:
Please do not use "JabRefPreferences" as whole. Please use a separate preference object. See https://jabref.readthedocs.io/en/latest/code-howtos/#how-to-work-with-preferences for pointers for more information.

Reply:
Fixed in commit 61ed7aea7ee49f9d06f075abffdefbecac39e73a